oppija.py

```diff
@@ -34,20 +45,73 @@ def opisCountHandler(self, bot, update, args=''):
         result = db.countOpis(update.message.chat.id)
         bot.sendMessage(chat_id=update.message.chat_id, text=(str(result[0]) + ' opis'))
 
-    def randomOppiHandler(self, bot, update):
-        result = db.randomOppi(update.message.chat.id)
-        bot.sendMessage(chat_id=update.message.chat_id, text=(result[0] + ': ' + result[1]))
+    def randomOppiHandler(self, bot, update, inverted=False):
+        if (inverted):
+            result = db.randomOppi(update.message.chat.id)
+            inverted_result = self.invertStringList(result)
+            bot.sendMessage(chat_id=update.message.chat_id, text=(inverted_result[1] + ' :' + inverted_result[0]))
+        else:
+            result = db.randomOppi(update.message.chat.id)
+            bot.sendMessage(chat_id=update.message.chat_id, text=(result[0] + ': ' + result[1]))
+
+    def invertStringList(self, list):
+        # Reference table for the Unicode chars: http://www.upsidedowntext.com/unicode
+        chars_standard = 'abcdefghijklmnopqrstuvwxyzåäö'
+        chars_inverted = 'ɐqɔpǝɟbɥıɾʞןɯuodbɹsʇnʌʍxʎzɐɐo'
+
+        chars_standard += '_,;.?!/\\\'<>(){}[]`&'
+        chars_inverted += '‾\'؛˙¿¡/\\,><)(}{][,⅋'
+
+        chars_standard += 'ABCDEFGHIJKLMNOPQRSTUVWXYZÅÄÖ'
+        chars_inverted += '∀qϽᗡƎℲƃHIſʞ˥WNOԀὉᴚS⊥∩ΛMXʎZ∀∀O'
```

Comment:
Have you checked that these chars render correctly on mobile devices?

Reply:
Yeah, they are standard Unicode chars. I tested them in the standard Telegram client and they work as intended. I would presume they also work in Telegram X.